chain/index/gc.go

```diff
@@ -75,17 +76,24 @@ func (si *SqliteIndexer) gc(ctx context.Context) {
 	// -------------------------------------------------------------------------------------------------
 	// Also GC eth hashes
 
-	// Convert gcRetentionEpochs to number of days
-	gcRetentionDays := si.gcRetentionEpochs / (builtin.EpochsInDay)
-	if gcRetentionDays < 1 {
-		log.Infof("skipping gc of eth hashes as retention days is less than 1")
-		return
-	}
+	// Calculate the retention duration based on the number of epochs to retain.
+	// retentionDuration represents the total duration (in nano seconds) for which data should be retained before considering it for garbage collection.
+	// graceDuration represents the additional duration (in nano seconds) to retain data after the retention duration.
+	// Since time.Duration expects a nanosecond value, we multiply the total seconds by time.Second to convert it to nanoseconds.
+	retentionDuration := time.Duration(si.gcRetentionEpochs*builtin.EpochDurationSeconds) * time.Second
+	graceDuration := time.Duration(graceEpochs*builtin.EpochDurationSeconds) * time.Second
+
+	// Calculate the total duration to retain data.
+	totalRetentionDuration := retentionDuration + graceDuration
+	currHeadTime := time.Unix(int64(head.MinTimestamp()), 0)
+	// gcTime is the time that is (gcRetentionEpochs + graceEpochs) in nano seconds before currHeadTime
+	gcTime := currHeadTime.Add(-totalRetentionDuration)
```

Comment:
if this is less than or equal to 0, return without doing anything. Also please can we had a test for when `gcTime ` <= 0 ?